Documentation for redesigned sidechains (#1516)
XRPLF/xrpl-dev-portal: content/references/http-websocket-apis/public-api-methods/ledger-methods/ledger_entry.md

- [`payment_channel`](#get-paychannel-object)
- [`deposit_preauth`](#get-depositpreauth-object)
- [`ticket`](#get-ticket-object)
- [`bridge`](#get-bridge-object)

> - [`xchain_claim_id`](#get-xchain-claim-id-object)

Reviewer: This is one of the broken links. The standard (GitHub-Flavored Markdown) ID formula does not replace underscores with dashes. So it _would_ be `#get-xchain_claim_id-object` based on the heading on line 620, except I think the proper name of this object is XChainClaimID, so it _should_ be:

```suggestion
- [`xchain_claim_id`](#get-xchainclaimid-object)
```

**Note:** This change assumes a corresponding change to the heading on on line 620.

> - [`xchain_create_account_claim_id`](#get-xchain-create-account-claim-id-object)

Reviewer: Similar to the `xchain_claim_id` change above.

```suggestion
- [`xchain_create_account_claim_id`](#get-xchaincreateaccountclaimid-object)
```

| ``          | String - [Address][]       | _(Required if `ticket` is specified as an object)_ The owner of the Ticket object. |

<!-- MULTICODE_BLOCK_START -->

*WebSocket*

```json
{
  "id": "example_get_bridge",
  "command": "ledger_entry",
  "bridge": {
    "issuing_chain_door": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt",
    "issuing_chain_issue": "XRP",
    "locking_chain_door": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC",
    "locking_chain_issue": "XRP"
  },
  "ledger_index": "validated"
}
```

*JSON-RPC*

```json
{
  "method": "ledger_entry",
  "params": [{
   "bridge": {
    "issuing_chain_door": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt",
    "issuing_chain_issue": "XRP",
    "locking_chain_door": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC",
    "locking_chain_issue": "XRP"
   },
  "ledger_index": "validated"
  }]
}
```

*Commandline*

```sh
rippled json ledger_entry '{ "bridge": { "issuing_chain_door": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt", "issuing_chain_issue": "XRP", "locking_chain_door": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC", "locking_chain_issue": "XRP" }, "ledger_index": "validated" }'
```

<!-- MULTICODE_BLOCK_END -->

> ### Get `xchain_claim_id`

Reviewer: `xchain_claim_id` is not a typical "canonical" name for a first-class ledger entry; typically the proper name is in PascalCase. The snake_case version is really only a shortcut used in the ledger_entry request format. Based on the example response below, the canonical ledger entry name is XChainClaimID.

Also, for consistency with the other headers in the page, add "Object" to the header.

```suggestion
### Get XChainClaimID Object
```

Author: Updated

> Retrieve a [Cross-chain Claim ID object](xchains.html). Can be provided as string (object ID of the cross-chain claim) or as an object. _(Added by the Sidechains amendment)_

Reviewer: This is the source of one of those broken links, BTW. There's no page `xchains.html`. Not sure if that's a file you forgot to add to the PR or if this is supposed to be something else.

Author: Have removed this link as it's not the correct one.

<!-- MULTICODE_BLOCK_START -->

*WebSocket*

```json
{
  "id": "example_get_xchain_claim_id",
  "command": "ledger_entry",
  "xchain_claim_id": {
    "issuing_chain_door": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt",
    "issuing_chain_issue": "XRP",
    "locking_chain_door": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC",
    "locking_chain_issue": "XRP",
    "xchain_claim_id":1
  },
  "ledger_index": "validated"
}
```

*JSON-RPC*

```json
{
  "method": "ledger_entry",
  "params": [{
   "xchain_claim_id": {
    "issuing_chain_door": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt",
    "issuing_chain_issue": "XRP",
    "locking_chain_door": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC",
    "locking_chain_issue": "XRP",
    "xchain_claim_id":1
  },
  "ledger_index": "validated"
  }]
}
```

*Commandline*

```sh
rippled json ledger_entry '{ "xchain_claim_id": { "issuing_chain_door": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt", "issuing_chain_issue": "XRP", "locking_chain_door": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC", "locking_chain_issue": "XRP", "xchain_claim_id":1 }, "ledger_index": "validated" }'
```

<!-- MULTICODE_BLOCK_END -->

> ### Get `xchain_create_account_claim_id`

Reviewer: Similar to above, I think this should be:

```suggestion
### Get XChainCreateAccountClaimID Object
```

Author: Updated

> The `xchain_create_account_claim_id` object is created when the sidechain is first notified (via an attestation) that a transaction occurred in the locking chain, and is removed when a quorum of attestations is reached. 

Reviewer: Object name should be in PascalCase here. Also, that's a lot passive voice.

> Retrieve a `xchain_create_account_claim_id` object which connects a locking chain and an issuing chain. Can be provided as string (object ID of the Bridge) or as an object. _(Added by the Sidechains amendment)_

Reviewer: As we discussed, maybe call it the "In-development Sidechains feature" or something along those lines and add the `:not_enabled:` icon to make it clearer that this is not even a proposed amendment yet.

<!-- MULTICODE_BLOCK_END -->

<!-- TODO: enable if/when Tickets are available on Mainnet
[Try it! >](websocket-api-tool.html#ledger_entry-ticket)
-->

### Get Bridge Object

> Retrieve a [Bridge object](bridge.html) which connects a locking chain and an issuing chain. Can be provided as string (object ID of the Bridge) or as an object. _(Added by the [Sidechains amendment][])_

Reviewer: The `[Sidechains amendment][]` shorthand won't work unless you add it to the `content/_snippets/rippled-api-links.md` file (there's a list of amendments in there).

Reviewer 2: It's also now called the `XChainBridge` amendment

Reviewer: It's also not an amendment in a released or version of the code so it doesn't make sense to list it officially (partly because of things like this where the names change before release).

I recommend using a disclaimer such as the [AMM disclaimer snippet](https://github.com/XRPLF/xrpl-dev-portal/blob/master/content/_snippets/amm-disclaimer.md) or the one in the [STIssue description](https://xrpl.org/serialization.html#issue-fields).

```suggestion
_(This is part of the proposed [Sidechains](https://dev.to/ripplexdev/xrp-ledger-sidechains-redesigned-190i) functionality :not_enabled:, and is subject to change.)_

Retrieve a [Bridge object](bridge.html) which connects a locking chain and an issuing chain. Can be provided as string (object ID of the Bridge) or as an object.
```

Reviewer: Also, It looks like `Bridge` is a new type of ledger entry that doesn't have its own page yet? Make sure to add that before you merge.

| `Field`                 | Type                       | Description           |
|:------------------------|:---------------------------|:----------------------|
| `bridge`                | Object or String           | The [[Bridge object](bridge.html) to retrieve. |

> | ``          | String - [Address][]       | _(Required if `ticket` is specified as an object)_ The owner of the Ticket object. |

Reviewer: Incomplete? The field name is missing and the description refers to tickets. You probably want four lines:

- `bridge.issuing_chain_door`
- `bridge.issuing_chain_issue`
- `bridge.locking_chain_door`
- `bridge.locking_chain_issue`

<!-- MULTICODE_BLOCK_START -->

*WebSocket*

```json
{
  "id": "example_get_bridge",
  "command": "ledger_entry",
  "bridge": {
    "issuing_chain_door": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt",
    "issuing_chain_issue": "XRP",
    "locking_chain_door": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC",
    "locking_chain_issue": "XRP"
  },
  "ledger_index": "validated"
}
```

*JSON-RPC*

```json
{
  "method": "ledger_entry",
  "params": [{
   "bridge": {
    "issuing_chain_door": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt",
    "issuing_chain_issue": "XRP",
    "locking_chain_door": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC",
    "locking_chain_issue": "XRP"
   },
  "ledger_index": "validated"
  }]
}
```

*Commandline*

```sh
rippled json ledger_entry '{ "bridge": { "issuing_chain_door": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt", "issuing_chain_issue": "XRP", "locking_chain_door": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC", "locking_chain_issue": "XRP" }, "ledger_index": "validated" }'
```

<!-- MULTICODE_BLOCK_END -->

### Get xchain_claim_id

> Retrieve a [Crosschain Claim ID object](xchains.html). Can be provided as string (object ID of the crosschain claim) or as an object. _(Added by the [Sidechains amendment][])_

Reviewer: Not sure what page `xchains.html` is supposed to be, but it doesn't exist yet in this PR, so this is a broken link.

    "issuing_chain_issue": "XRP",
    "locking_chain_door": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC",
    "locking_chain_issue": "XRP",
    "xchain_claim_id":1
  },
  "ledger_index": "validated"
  }]
}
```

*Commandline*

```sh
rippled json ledger_entry '{ "xchain_claim_id": { "issuing_chain_door": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt", "issuing_chain_issue": "XRP", "locking_chain_door": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC", "locking_chain_issue": "XRP", "xchain_claim_id":1 }, "ledger_index": "validated" }'
```

<!-- MULTICODE_BLOCK_END -->

### Get xchain_create_account_claim_id

The `xchain_create_account_claim_id` object is created when the sidechain is first notified (via an attestation) that a transaction occurred in the locking chain, and is removed when a quorum of attestations is reached. 

Retrieve a [xchain_create_account_claim_id object](bridge.html) which connects a locking chain and an issuing chain. Can be provided as string (object ID of the Bridge) or as an object. _(Added by the [Sidechains amendment][])_

| `Field`                 | Type                       | Description           |
|:------------------------|:---------------------------|:----------------------|

> | `bridge`                | Object or String           | The [[Bridge object](bridge.html) to retrieve. |

Reviewer: I think this is supposed to be `xchain_create_account_claim_id`? (Boy, what a mouthful)

> | ``          | String - [Address][]       | _(Required if `ticket` is specified as an object)_ The owner of the Ticket object. |

Reviewer: Don't forget to finish this before publishing

<!-- MULTICODE_BLOCK_START -->

*WebSocket*

```json
{

>   "id": "example_get_bridge",

Reviewer: If you're going to use meaningful IDs like this, make sure they accurately describe the request

<!-- MULTICODE_BLOCK_END -->

> ### Response for Bridge Object

Reviewer: This section is unnecessary since the response format is the same for all object types (except the contents of the `node` field, which are always the canonical object definition). So you should probably delete all the changes below here. (The example objects may be useful when you get to adding these to the [ledger entries (object types) docs](https://xrpl.org/ledger-object-types.html).

Author: As discussed, deleted the additional examples from this section

      "index": "DC74C41BDD449D5DC421E521E69CB903F3A4A97C9C12BD638E055BF46B1EF367"
    },
    "status": "success",
    "validated": false
  }
}
```

*Commandline*

```json
{
  "result": {
    "index": "DC74C41BDD449D5DC421E521E69CB903F3A4A97C9C12BD638E055BF46B1EF367",
    "ledger_current_index": 4,
    "node": { 
      "Account": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC",
      "Balance": "0",
      "Flags": 0,
      "LedgerEntryType": "Bridge",
      "MinAccountCreateAmount": "20000000",
      "OwnerNode": "0",
      "PreviousTxnID": "C73BBB7F7B22A5C27217597CADA505462E1F5DEABB32CF23C2D72BC4D36225EF",
      "PreviousTxnLgrSeq": 3,
      "SignatureReward": "1000000",
      "XChainAccountClaimCount": "0",
      "XChainAccountCreateCount": "0",
      "XChainBridge": {
        "IssuingChainDoor": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt",
        "IssuingChainIssue": "XRP",
        "LockingChainDoor": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC",
        "LockingChainIssue": "XRP"
      },
      "XChainClaimID": "0",
      "index": "DC74C41BDD449D5DC421E521E69CB903F3A4A97C9C12BD638E055BF46B1EF367"
    },
    "status": "success",
    "validated": false
  }
}
```

<!-- MULTICODE_BLOCK_END -->

>  ### Response for XChain Claim ID

Reviewer: The space at the start of this line is making it not render as a header.

```suggestion
### Response for XChain Claim ID
```

But that's kind of a moot point since you can delete this whole example responses section.

Author: As discussed, deleted the example responses section.

        "IssuingChainIssue": "XRP",
        "LockingChainDoor": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC",
        "LockingChainIssue": "XRP"
      },
      "XChainClaimID": "0",
      "index": "DC74C41BDD449D5DC421E521E69CB903F3A4A97C9C12BD638E055BF46B1EF367"
    },
    "status": "success",
    "validated": false
  }
}
```

<!-- MULTICODE_BLOCK_END -->

 ### Response for XChain Claim ID

{"index": "B459014123A6535823584F96DB50721FE517FE697B7E21275A84C43FACBF9942",
 "ledger_current_index": 6,
 "node": 
 {"Account": "r9A8UyNpW3X46FUc6P7JZqgn6WgAPjBwPg",
  "Flags": 0,
  "LedgerEntryType": "XChainClaimID",
  "OtherChainAccount": "rnJmYAiqEVngtnb5ckRroXLtCbWC7CRUBx",
  "OwnerNode": "0",
  "PreviousTxnID": "29B3F0A2FD3A7765CBFDA15CC91B11BF6A1B574F1BE43611D7C33EF72353C000",
  "PreviousTxnLgrSeq": 4,
  "SignatureReward": "1000000",
  "XChainBridge": 
  {"IssuingChainDoor": "rKeSSvHvaMZJp9ykaxutVwkhZgWuWMLnQt",
   "IssuingChainIssue": "XRP",
   "LockingChainDoor": "rJvExveLEL4jNDEeLKCVdxaSCN9cEBnEQC",
   "LockingChainIssue": "XRP"},
  "XChainClaimAttestations": [],
  "XChainClaimID": "1",
  "index": "B459014123A6535823584F96DB50721FE517FE697B7E21275A84C43FACBF9942"},
 "status": "success",

>  "validated": false}

Reviewer: This should probably be in a code block? Maybe on _in_ a tab?
